core/vmpu/src/mpu_armv8m/vmpu_armv8m_unpriv_access.c

```diff
@@ -54,73 +55,32 @@
-    uint32_t test_addr_upper = (addr + size - 1) & ~31UL;
-
-    /* Test lower address. */
-    asm volatile (
-        "tta %[response], %[addr]"
-        : [response] "=r" (response_lower)
-        : [addr] "r" (test_addr_lower)
-    );
-    if (test_addr_lower != test_addr_upper) {
-        /* Test upper address. */
-        asm volatile (
-            "tta %[response], %[addr]"
-            : [response] "=r" (response_upper)
-            : [addr] "r" (test_addr_upper)
-        );
-        /* If lower and upper do not have the same S|SRVALID|SREGION, then it's definitely not the same region. */
-        if (((response_lower ^ response_upper) & (TT_RESP_S_Msk | TT_RESP_SRVALID_Msk | TT_RESP_SREGION_Msk))) {
-            /* Upper memory region has different SAU region than lower memory region! */
-            return 0;
-        }
-        /* Both memory locations have the same non-secure SAU region and therefore same properties.
-         * No Secure SAU region can be inbetween due to SAU region overlap rules. */
-        response_lower &= response_upper;
-    }
-
-    return response_lower & (TT_RESP_NSRW_Msk | TT_RESP_NSR_Msk | TT_RESP_RW_Msk | TT_RESP_R_Msk |
-                             TT_RESP_S_Msk | TT_RESP_SRVALID_Msk | TT_RESP_SREGION_Msk);
-}
-
-extern int vmpu_fault_recovery_mpu(uint32_t pc, uint32_t sp, uint32_t fault_addr, uint32_t fault_status);
-
 uint32_t vmpu_unpriv_access(uint32_t addr, uint32_t size, uint32_t data)
 {
-    unsigned int tries = 0;
-    while(1) {
-        if ((vmpu_unpriv_test_range(addr, UVISOR_UNPRIV_ACCESS_SIZE(size)) & (TT_RESP_NSRW_Msk | TT_RESP_SRVALID_Msk)) == (TT_RESP_NSRW_Msk | TT_RESP_SRVALID_Msk)) {
-            switch(size) {
-                case UVISOR_UNPRIV_ACCESS_READ(1):
-                    return *((uint8_t *) addr);
-                case UVISOR_UNPRIV_ACCESS_READ(2):
-                    return *((uint16_t *) addr);
-                case UVISOR_UNPRIV_ACCESS_READ(4):
-                    return *((uint32_t *) addr);
-                case UVISOR_UNPRIV_ACCESS_WRITE(1):
-                    *((uint8_t *) addr) = (uint8_t) data;
-                    return 0;
-                case UVISOR_UNPRIV_ACCESS_WRITE(2):
-                    *((uint16_t *) addr) = (uint16_t) data;
-                    return 0;
-                case UVISOR_UNPRIV_ACCESS_WRITE(4):
-                    *((uint32_t *) addr) = data;
-                    return 0;
-                default:
-                    break;
-            }
-            break;
-        }
-        if (++tries > 1 || !vmpu_fault_recovery_mpu(0, 0, addr, 0)) {
-            break;
+    //if ((vmpu_unpriv_test_range(addr, UVISOR_UNPRIV_ACCESS_SIZE(size)) & (TT_RESP_NSRW_Msk | TT_RESP_SRVALID_Msk)) == (TT_RESP_NSRW_Msk | TT_RESP_SRVALID_Msk)) {
+    //this should be slower substantualy
+    if (vmpu_buffer_access_is_ok(g_active_box, (const void *) addr, UVISOR_UNPRIV_ACCESS_SIZE(size))){
+        switch(size) {
```

Comment:
Add a space after keywords like `switch`, before the `(`. It was wrong before. Please make style changes in a separate commit.